src/Illuminate/Foundation/Console/RouteCacheCommand.php

```diff
@@ -98,12 +101,16 @@ protected function getFreshApplication()
      * Build the route cache file.
      *
      * @param  \Illuminate\Routing\RouteCollection  $routes
+     * @param  array  $compiledRoutes
      * @return string
      */
-    protected function buildRouteCacheFile(RouteCollection $routes)
+    protected function buildRouteCacheFile(RouteCollection $routes, array $compiledRoutes)
     {
         $stub = $this->files->get(__DIR__.'/stubs/routes.stub');
 
-        return str_replace('{{routes}}', base64_encode(serialize($routes)), $stub);
+        $replaced = str_replace('{{routes}}', base64_encode(serialize($routes)), $stub);
+        $replaced = str_replace('{{compiledRoutes}}', base64_encode(serialize($compiledRoutes)), $replaced);
```

Comment:
Is it possible to use raw php instead of base64+serialize?
I'd guess the cost of decoding+unserialing completely dominates the matching time, and this is why things are not faster?

Reply:
@nicolas-grekas I actually also tested with the dumped PHP array file from the `CompiledUrlMatcherDumper` but that didn't lead to better results. 

On that specific case: I had to use reflection to access the `generateCompiledRoutes` method so I could combine the content with the compiled routes from Laravel. I could create two separate cache files but I'd like to keep everything together in one if possible. Any chance we can make the `generateCompiledRoutes` method public or extract the contents to a new public method?